src/complex/Utilities/DataArrayUtilities.hpp

```diff
@@ -904,4 +906,215 @@ class CopyTupleUsingIndexList
   IDataArray& m_NewCellArray;
   nonstd::span<const int64> m_NewIndices;
 };
+
+/**
+ * @brief This class is meant to make copying data from one IArray into another easier for the developer.
+ *
+ * An example use of these functions would be the following (where newCellData is an AttributeMatrix in dataStructure ):
+ *   ParallelTaskAlgorithm taskRunner;
+ *   for (const auto& [dataId, dataObject] : *newCellData)
+ *   {
+ *     auto* inputDataArray = dataStructure.getDataAs<IArray>(inputCellDataPath.createChildPath(name));
+ *     auto* destDataArray = dataStructure.getDataAs<IArray>(destCellDataPath.createChildPath(name));
+ *     auto* newDataArray = dataStructure.getDataAs<IArray>(newCellDataPath.createChildPath(name));
+ *     const IArray::ArrayType arrayType = destDataArray->getArrayType();
+ *     CopyFromArray::RunParallel<CopyFromArray::Combine>(arrayType, destDataArray, taskRunner, inputDataArray, newDataArray);
+ *   }
+ *   taskRunner.wait();
+ * @code
+ *
+ * @endcode
+ */
+class CopyFromArray
+{
+public:
+  CopyFromArray()
+  {
+  }
+  ~CopyFromArray() = default;
+
+  CopyFromArray(const CopyFromArray&) = default;
+  CopyFromArray(CopyFromArray&&) noexcept = default;
+  CopyFromArray& operator=(const CopyFromArray&) = delete;
+  CopyFromArray& operator=(CopyFromArray&&) noexcept = delete;
+
+  template <bool UseCombine, bool UseAppend>
+  struct TemplateTypeOptions
+  {
+    static inline constexpr bool UsingCombine = UseCombine;
+    static inline constexpr bool UsingAppend = UseAppend;
+  };
+
+  using Combine = TemplateTypeOptions<true, false>;
+  using Append = TemplateTypeOptions<false, true>;
+
+  template <class TemplateTypeOptions = Combine, class ParallelRunnerT, class... ArgsT>
+  static void RunParallel(IArray::ArrayType arrayType, IArray* destArray, ParallelRunnerT&& runner, ArgsT&&... args)
```

Comment:
If combine and append cannot be used together, I feel like it would make sense to keep them separate.

Reply:
@JDuffeyBQ @nyoungbq and I went back and forth on that one. We were trying to avoid duplicating more code but I can an argument either way.

Reply:
Is it possible to have two functions that share most of the important code in one function?